### PR TITLE
Add doc-gen tool to generate documentation

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

**Comment by instagibbs** [comment]
nice! Concept ACK

**Comment by wintercooled** [comment]
tACK - I will merge other main header items after this is merged and reinstate community lop level link.

**Review by smoovhogg12 — APPROVED** [review]

**Pull request merged**
